src/main/java/org/folio/circulation/domain/validation/UniqRequestValidator.java

```diff
@@ -0,0 +1,49 @@
+package org.folio.circulation.domain.validation;
+
+import static org.folio.circulation.domain.RequestStatus.OPEN_AWAITING_PICKUP;
+import static org.folio.circulation.domain.RequestStatus.OPEN_IN_TRANSIT;
+import static org.folio.circulation.domain.RequestStatus.OPEN_NOT_YET_FILLED;
+
+import java.util.EnumSet;
+import java.util.Objects;
+import java.util.function.Function;
+
+import org.folio.circulation.domain.Request;
+import org.folio.circulation.domain.RequestAndRelatedRecords;
+import org.folio.circulation.domain.RequestStatus;
+import org.folio.circulation.support.HttpFailure;
+import org.folio.circulation.support.Result;
+
+public class UniqRequestValidator {
+
+  private static final EnumSet<RequestStatus> OPEN_STATUS = EnumSet.of(
+    OPEN_AWAITING_PICKUP,
+    OPEN_NOT_YET_FILLED,
+    OPEN_IN_TRANSIT
+  );
+
+  private final Function<RequestAndRelatedRecords, ? extends HttpFailure> errorSupplier;
+
+  public UniqRequestValidator(Function<RequestAndRelatedRecords, ? extends HttpFailure> errorSupplier) {
+    this.errorSupplier = errorSupplier;
+  }
+
+  public Result<RequestAndRelatedRecords> refuseWhenRequestIsAlreadyExisted(RequestAndRelatedRecords request) {
+    return Result.of(() -> request)
+      .failWhen(this::isRequesterHasAlreadyRequestForTheItem, errorSupplier::apply);
+  }
+
+  private Result<Boolean> isRequesterHasAlreadyRequestForTheItem(RequestAndRelatedRecords request) {
+    return Result.of(() ->
+      request
+        .getRequestQueue()
+        .getRequests()
+        .stream()
+        .anyMatch(it -> isTheSameRequester(request, it) && OPEN_STATUS.contains(it.getStatus()))
```

Comment:
Request class has method isOpen(), likely no need to create duplicates for request statuses

Reply:
Thanks I didn't know about it. 
isOpen method has 2 of 3 statuses OPEN_AWAITING_PICKUP, OPEN_NOT_YET_FILLED 
I need OPEN_IN_TRANSIT
@marc-johnson Shell I add OPEN_IN_TRANSIT to the isOpen method and then I will be able to use this method here?

Reply:
If isOpen() can be extended for this logic check, is it possible to remove this validator and follow the same pattern of CreateRequestService::refuseWhenItemIsNotValid to reduce overhead and complexity? I'm just seeing lots of wrappers here around the check and I wonder if they're necessary.

Reply:
Done

Reply:
@Anatolii-Starkov  The validator is currently still a separate class, does that mean that you disagree with @mtraneis suggestion to move it into the create request service, given that it is only used there?

Reply:
No it doesn't. As you can see I need to implement this story one more time. So I just made atomic change and mark it just for me 😄 
I had a plan to remove separate validator next day. 
Now all comments are fixed